carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/application/ParticipantService.kt

```diff
@@ -0,0 +1,60 @@
+package dk.cachet.carp.studies.application
+
+import dk.cachet.carp.common.EmailAddress
+import dk.cachet.carp.common.UUID
+import dk.cachet.carp.studies.domain.ParticipantGroupStatus
+import dk.cachet.carp.studies.domain.users.AssignParticipantDevices
+import dk.cachet.carp.studies.domain.users.Participant
+
+
+/**
+ * Application service which allows adding participants to studies and creating deployments for them.
+ */
+interface ParticipantService
+{
+    /**
+     * Add a [Participant] to the study with the specified [studyId], identified by the specified [email] address.
+     * In case the [email] was already added before, the same [Participant] is returned.
+     *
+     * @throws IllegalArgumentException when a study with [studyId] does not exist.
+     */
+    suspend fun addParticipant( studyId: UUID, email: EmailAddress ): Participant
```

Comment:
We do currently support usernames instead of emails. Is the endpoint for adding username participants deliberately left out?

Reply:
I guess so. Afaik, it was never there before.